libs/messaging/messaging-impl/src/main/kotlin/net/corda/messaging/mediator/RPCClient.kt

```diff
@@ -67,10 +71,17 @@ class RPCClient(
     }
 
     private fun buildHttpRequest(message: MediatorMessage<*>): HttpRequest {
-        return HttpRequest.newBuilder()
+
+        val builder = HttpRequest.newBuilder()
             .uri(URI(message.endpoint()))
             .POST(HttpRequest.BodyPublishers.ofByteArray(message.payload as ByteArray))
-            .build()
+
+        // Add key HTTP header
+        message.getPropertyOrNull(MSG_PROP_KEY)?.let { key ->
+            builder.header("corda-request-key", key.toString())
+        }
```

Comment:
```suggestion
        message.getPropertyOrNull(MSG_PROP_KEY)?.let { value ->
            builder.header("corda-request-key", value.toString())
        }
```

Also, might be a good idea to introduce a constant for `"corda-request-key"` in case we need it from elsewhere in the code. E.g. in the unit test.